Cross-platform support, completely refactor the project.

PurelyAndy opened the pull request:
I'm rewriting the GUI with Avalonia and restructuring the projects, moving shared code, cleaning up, etc.

![image](https://github.com/user-attachments/assets/44c38c48-7844-4f56-bc4f-bf819cc6e10f)

PurelyAndy committed:
GUI is mostly done, NOR-reading code is added but untested

PurelyAndy committed:
COM Ports not found hard crash fix

Co-Authored-By: VashBaldeus <vashbaldeus@gmail.com>

ScrubN commented on PS5 NOR Modifier/NORInfo.cs:
```diff
@@ -0,0 +1,12 @@
+
+public class NORData
+{
+    public string? OffsetOne; // TODO: These two need better names
+    public string? OffsetTwo;
+    public string? WiFiMAC;
+    public string? LANMAC;
+    public string? Serial;
+    public string? Variant;
+    public string? MoboSerial;
```
These should remain serialized as `byte[]`s, or even better the entire NOR should be read into memory and wrapped by a class with functions for extracting this information as `byte[]`s/`ReadOnlySpan<byte>`s.

PurelyAndy committed:
Fix #9 (.BIN instead of .bin extension)

Co-Authored-By: stevebroskey <946790+stevebroskey@users.noreply.github.com>

PurelyAndy committed:
Remove files that shouldn't be tracked and unnecessary NORInfo class

PurelyAndy committed:
Make NOR reading code good

PurelyAndy committed:
the person who wrote this program should know that it upsets me greatly.

anyway, reading and writing are done.

PurelyAndy committed:
Untested UART code.

PurelyAndy commented:
I don't own a PS5, so I can't test the UART communication code. If anybody is willing to help me out with that, please let me know.

PurelyAndy committed:
Remove spaces from the project name

PurelyAndy committed:
Improve README

PurelyAndy committed:
Start moving shared code into one project

falconmick commented:
Don’t have an open ps5 to test on but lmk if you would like any help

benjamin238 commented:
Hey @PurelyAndy, I've taken a brief look at the refactored code and agree that this is the direction we should be heading in. This fork is very well organized and solves many problems the current main branch has. 

I'm attempting to merge a small change I made but I don't have write access to your fork. Would you be able to provide that for me? Thanks :3

Ayymoss commented:
This is good. We still need to clean up https://github.com/PurelyAndy/PS5NorModifier/blob/main/UART-CL_By_TheCod3r/Program.cs's monolithic class

PurelyAndy commented:
Yeah, I'm working on that right now. There's a lot of shared code between the GUI and CLI apps that needs to be consolidated.

PurelyAndy committed:
Merge branch 'main' into main

PurelyAndy commented:
@benjamin238 I'll give you access now.

PurelyAndy committed:
More work on cleaning up UART-CL.

PurelyAndy committed:
Move UART and NOR pages into their own controls.

PurelyAndy committed:
Downgrade to .NET 8 and refactor UART GUI, moving code into UARTLib

PurelyAndy committed:
UART code in UART-CL now uses UARTLib

PurelyAndy committed:
Rename UARTLib to PS5Lib, because it doesn't only do UART stuff.

Someone please give me a better name for this.

PurelyAndy committed:
UART-CL just needs to be cleaned up now.

Ayymoss commented:
Why downgrade from .NET 9 to .NET 8 out of curiosity? 

Could it not be compiled as Native AOT? Avalonia supports it: https://docs.avaloniaui.net/docs/deployment/native-aot

PurelyAndy commented:
.NET 8 is LTS. I don't know how often they want to update the .NET version of this project. It's trivial to update it back to .net ( or 10 in the future, anyway.

benjamin238 committed:
Internal Documentation, Refactor GetData, Fix Dependencies

Internally document model codes, rename some variables with unclear names, resolved dependency problems (wouldn't compile without this)

PurelyAndy committed:
work

PurelyAndy committed:
Add motherboard serial number editor to GUI, since UART-CL allows it

PurelyAndy committed:
Rename stuff, fix error db download

PurelyAndy committed:
Start abstracting CLI menus

PurelyAndy committed:
Finish refactoring the CLI app

xxbiohazrdxx commented:
I've also submitted a PR for a rewrite and to clean up the CLI, but I took a more drastic approach. @PurelyAndy if you want to take a look #33

PurelyAndy committed:
Add some documentation, make some names more accurate

PurelyAndy committed:
Merge branch 'main' into main
